app/pods/agendas/template.hbs

```diff
@@ -19,12 +40,176 @@
+    @message={{t "filtered-agendas"}}
+    @buttonText={{t "clear-filter"}}
+    @clearAction={{this.clearFilter}}
+  />
+{{/if}}
+<div class="auk-scroll-wrapper">
+  <div class="auk-scroll-wrapper__body">
+    {{#if @model.length}}
+      <div class="auk-u-ml-4">
+        <table
+          data-test-route-agendas-overview-data-table
+          class="auk-table auk-table--agenda auk-table--sticky-header"
+        >
+          <thead>
+            <tr>
+              <Utils::ThSortable
+                class="auk-table__col--3 auk-table__header-title"
+                @currentSorting={{this.sort}}
+                @field="created-for.planned-start"
+                @onChange={{this.sortTable}}
+                @label={{t "agenda"}}
+              />
+              <Utils::ThSortable
+                class="auk-table__header-title"
+                @currentSorting={{this.sort}}
+                @field="status"
+                @onChange={{this.sortTable}}
+                @label={{t "status"}}
+              />
+              <Utils::ThSortable
+                class="auk-table__header-title"
+                @currentSorting={{this.sort}}
+                @field="created-for.kind.label"
+                @onChange={{this.sortTable}}
+                @label={{t "kind"}}
+              />
+              <Utils::ThSortable
+                class="auk-table__header-title"
+                @currentSorting={{this.sort}}
+                @field="modified"
+                @onChange={{this.sortTable}}
+                @label={{t "latest-modified"}}
+              />
+              <th></th>
+            </tr>
+          </thead>
+          <tbody>
+            {{#if this.isLoadingModel}}
+              <tr>
+                <td
+                  data-test-route-agendas-overview-loader
+                  colspan="5"
+                >
+                  {{t "loading"}}
+                </td>
+              </tr>
+            {{else}}
+              {{#each @model as |agenda|}}
+                <tr
+                  class="auk-u-text-align--center auk-table--interaction-cursor auk-table--clickable-rows"
+                  {{! template-lint-disable no-invalid-interactive }}
+                  {{on "click" (fn this.onClickRow agenda)}}
+                >
+                  <td data-test-route-agendas-overview-row-title>
+                    <span class="auk-o-flex auk-o-flex--vertical-center auk-o-flex--spaced">
+                      {{#if agenda.status.isDesignAgenda}}
+                        <strong>{{t "design-agenda"}}</strong>
+                        <Auk::Pill @skin="active-future">{{agenda.name}}</Auk::Pill>
+                      {{else}}
+                        <strong>{{t "agenda"}}</strong>
+                        <Auk::Pill>{{agenda.name}}</Auk::Pill>
+                      {{/if}}
+                      <span>–</span>
+                      <span>
+                        {{moment-format agenda.createdFor.plannedStart "DD.MM.YYYY"}}
```

Comment:
tried it out with 1100 agendas, that goes up to 8 seconds. not as bad as I expected, but frankly anything above 1 second is not great.

imo we can tackle this
- by making mu-cl-resources and/or mu-authorization faster (maybe this isn't actually feasible though)
- by creating a microservice that fetches the agendas. maybe the handwritten SPARQL query is faster than mu-cl-resource?
- taking the hit of lazy loading, but using skeleton loaders (or a different visual mechanism) to make clear to the user that we're still loading data, but that they can interacting with a part of the site

Reply:
I think this is a consequence of querying for agendas instead of meetings. There are just way more of them.
But this should improve with the addition of the filter on `has-no:next-version`?

Reply:
Also, we should only load the 20 or so agendas that are shown with page sizes, what do you mean "tried it out with 1100 agendas"?

Reply:
testing was done using `'filter[:has-no:next-version]': true,` as part of the query params. i also just created 1000 separate agendas (as if the "create new agenda" & modal was filled out 1000 times), so it would've been the same if we were querying for meetings or agendas (same amount of each, though in practice we'll have more agendas than meetings)

Reply:
> Also, we should only load the 20 or so agendas that are shown with page sizes, what do you mean "tried it out with 1100 agendas"?

I created 1100 agenda's and then queryed normally. As there's more data in the database, querying gets slower.

Reply:
"Because we're filtering and sorting, the DB has to deal with the whole dataset."
We are using pagination, so unless you are loading the page with `/?size=1100` it shouldn't take that long to load 10 agendas (default page size)?
The problem with the current code is that we are showing **wrong/missing data** between the initial load and when everything is finally loaded.
also a spam of seperate calls for each of that data is happening, which will also impact performance with larger page sizes and can put more stress on the backend.
![image](https://user-images.githubusercontent.com/22245223/164186849-acf246d9-efd7-420d-8cd5-7f6f24da246b.png)

Reply:
![loaded](https://user-images.githubusercontent.com/22245223/164187877-16c53229-19a4-4ead-8f8b-a91fe5a55eaf.png)
![still loading](https://user-images.githubusercontent.com/22245223/164187884-52e17b4d-426e-4fb7-9271-b796d83cc9b9.png)

Reply:
Yes, we're doing pagination, but even if we're only receiving 10 agendas, the DB still needs to go over all the agendas it has to know which 10 agendas to give us. It needs to remove all agendas which do not match our filtering criteria from the result set, and then it needs to go over all those agendas to sort them according to the sort options we requested.

If we were only requesting 10 agendas, no sorting or filtering involved, then yes this would be fast and the total amount of agendas in the backend wouldn't matter. But because we're filtering and sorting on top of the pagination, as the amount of data grows, the query will get slower.

I agree that the extra calls aren't ideal either. But it's a trade-off between 1 call that is slow (in our case, very slow), or a number of calls to lazy-load data.

I also agree that showing wrong data while it is loading is bad. We could use [skeleton loading](https://betterprogramming.pub/the-what-why-and-how-of-using-a-skeleton-loading-screen-e68809d7f702) for that, showing no data, and instead a grey square. This way the user knows data is still loading while still seeing something.

Reply:
It's ok if the loading of kind happens after the data is shown, that is also what happens in the current implementation.
My main problem is that we are showing wrong dates and that we are showing "agenda" before we fully know the status so it changes to "ontwerpagenda" later.

The main gripe is that we should show no data instead of wrong data.

Reply:
" But it's a trade-off between 1 call that is slow (in our case, very slow), or a number of calls to lazy-load data."

The agendas query in question is not that complex, currently there are less than 3000 agendas on prod DB.
Even with filtering and sorting, you would expect a fast query.
Either we are querying wrong or there is a problem in the backend.

Reply:
@sergiofenoll When adding > 1000 agenda's in the triplestore, are you running Virtuoso with the default config or with a config tuned for that amount of data [like on production](https://github.com/kanselarij-vlaanderen/app-kaleidos/blob/development/config/db/virtuoso.production.ini)? Without that tuned config the timings you put here are not representative.

When I run this branch with a copy of the ACC DB (incl. config), containing > 2000 agenda's, the query as it is now takes 2s on my machine with page size 25 and a cold cache and cold mu-cl-resource. Additional data is loaded in parallel in the magnitude of 1-2s. I don't see wrong data appear before the right data.

It is known that include is a very expensive operation in mu-cl-resources (there are plans to improve this in the future).  For smaller page sizes I don't see a problem in the additional requests. Hannes' opinion about the page size options (which I agree on) is that we should not offer page size options that we now will be too slow to be used in practice. So, regardless we go for include or not, I would remove page size options > 100.

Reply:
didn't realize there was a different config for production, also didn't realize it'd made such a big difference. i tried it using the dev configuration.

for what it's worth, the numbers i shared were also with a cold cache and cold mu-cl-resources, but *with* `include: 'status,created-for,created-for.kind'`, which is not what's currently committed.

Reply:
Right now `include: 'status,created-for,created-for.kind'` is committed. The query fetching the agendas is somewhat slow because of this, but we get the expected data from the first load.

What's the preferred solution in this case, live with the somewhat  slow query, or lazy load the agenda's meeting?

Reply:
Go with it for now, we can evaluate on ACC with larger agendas on a faster server.